presto-spi/src/main/java/com/facebook/presto/spi/type/CharType.java

```diff
@@ -94,7 +95,7 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
         if (leftLength != rightLength) {
             return false;
         }
-        return leftBlock.equals(leftPosition, 0, rightBlock, rightPosition, 0, leftLength);
+        return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
```

Comment:
Why did you need to change the implementation of `equalTo`, and does `hash` need to change?